pkg/cmd/cli/cmd/rollout/latest.go

```diff
@@ -0,0 +1,161 @@
+package rollout
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/meta"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+const (
+	rolloutLatestLong = `
+Start a new rollout for a deployment config with the latest state from its triggers
+
+This command is appropriate for running manual rollouts. If you want full control over
+running new rollouts, use "oc set triggers --manual" to disable all triggers in your
+deployment config and then whenever you want to run a new deployment process, use this
+command in order to pick up the latest images found in the cluster that are pointed by
+your image change triggers.`
+
+	rolloutLatestExample = `  # Start a new rollout that picks up the latest images found under the tag that your image change triggers use.
+  %[1]s rollout latest dc/nginx
+`
+)
+
+// RolloutLatestOptions holds all the options for the `rollout latest` command.
+// TODO: Support --dry-run
+type RolloutLatestOptions struct {
+	mapper meta.RESTMapper
+	typer  runtime.ObjectTyper
+	infos  []*resource.Info
+
+	out         io.Writer
+	shortOutput bool
+
+	oc              client.Interface
+	kc              kclient.Interface
+	baseCommandName string
+}
+
+// NewCmdRolloutLatest implements the oc rollout latest subcommand.
+func NewCmdRolloutLatest(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	opts := &RolloutLatestOptions{
+		baseCommandName: fullName,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "latest DEPLOYMENTCONFIG",
+		Short:   "Start a new rollout for a deployment config with the latest state from its triggers",
+		Long:    rolloutLatestLong,
+		Example: fmt.Sprintf(rolloutLatestExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			err := opts.Complete(f, cmd, args, out)
+			kcmdutil.CheckErr(err)
+
+			if err := opts.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+
+			err = opts.RunRolloutLatest()
+			kcmdutil.CheckErr(err)
+		},
+	}
+
+	kcmdutil.AddOutputFlagsForMutation(cmd)
+
+	return cmd
+}
+
+func (o *RolloutLatestOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	if len(args) != 1 {
```

Comment:
can we support multiple deployments? or is this limitation because we want to add `--follow` or `--logs` ?